### PR TITLE
Add sort command to sort across all candidate fields (Issue #72)

### LINKED ISSUE #72: Add `sort` command
Add `sort` command of the format `sort s/ATTRIBUTE_FIELD` to allow users to sort the list of TA candidates by a specific attribute field. The sorting will utilise the in-built java comparator for strings and numeric values.

### PR DESCRIPTION
**Pull request by charmainehly**
Currently, the default list is loaded from the json file.
There is no functionality to reorder this list.

User is not able to organise the list in a specific numerical
or alphabetical order.

Let's create a new sort command, which parses in the field
that the user wants to sort the candidate list by. The list
will order candidates based on the in-built java comparator
on the particular field.

Closes #72

### CODE REVIEW

**Comment by charmainehly** [comment]
Let me make a new commit to address the code coverage issues and test the relevant code! :)

File: src/main/java/seedu/address/logic/commands/SortCommand.java
Comment (domlimm) [inline src/main/java/seedu/address/logic/commands/SortCommand.java:16]: Very clean and neat codes! Good that the java docs/comments are present, also following the coding standard closely! 👍🏻

File: src/main/java/seedu/address/logic/parser/SortCommandParser.java
Comment (domlimm) [inline src/main/java/seedu/address/logic/parser/SortCommandParser.java:51]: Great usage of lambdas!

File: src/test/java/seedu/address/logic/parser/SortCommandParserTest.java
Comment (domlimm) [inline src/test/java/seedu/address/logic/parser/SortCommandParserTest.java:16]: Great extensive testing for our new `sort` feature!

**Review by domlimm — COMMENTED** [review]
Alright, everything looks good to me! Great job!

@LeeEnEn maybe can put this review on hold first. Am adding more stuff to this PR. Will let you know when to review, approve and merge in once done. Thanks!

Thanks @charmainehly! 💯

**Comment by domlimm** [comment]
@charmainehly thanks for adding in the refactor of `studentId`! 😄

**Review by LeeEnEn — APPROVED** [review]
LGTM! Super neat and clean! 👍👍

**Pull request merged**
